### PR TITLE
feat: Scene level lods

### PR DESCRIPTION
**Pull request by dalkia**
# Changes 

Adds scenes-level LODS

# How to test

- Download this LODS assets (https://drive.google.com/file/d/1bZ-J0jHu7iyjA_rliaulAOfv9gyQ4eg3/view?usp=sharing) and put them in the folder `StandaloneWindows64\StandaloneWindows64_Data\StreamingAssets\AssetBundles\lods` (you need to create the folder)
- Enter the world and check that you see scene LODs in the distance
- Walk to the border of the load radius and check that new scenes are loaded (different than the empty ones)
- In debug settings, click the `Toggle LOD color` button. Check that lod color changes

### CODE REVIEW

File: Explorer/Assets/DCL/LOD/LODAsset.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+namespace DCL.LOD
+{
+    public struct LODAsset : IDisposable
+    {
+        public LODKey LodKey;
+        public GameObject Root;
+        public AssetBundleData AssetBundleReference;
+        private readonly bool LoadingFailed;
+
+        private readonly LODDebugInfo LODDebugInfo;
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/LOD/LODAsset.cs:18]: I propose you to detach `Debugging Logic` from the prod-ready ready:

- It is too tightly coupled with the prod logic
- `LODDebugInfo` is a class
- `ILODSettingsAsset` does not depend on the instance: it should not be stored in the component

Expectations:
- `LODDebugInfo` can create an overhead only if `IsColorDebuging` is true

My suggestions:
- Move `LODDebugInfo` to a separate component
- Disconnect from `LODAsset`
- Handle it in a static `Utility` class/a separate system/separate branch of the logic
   - E.g. you can move `sceneLODInfo.IsDirty = false;` to the later system and introduce a system between `UpdateSceneLODInfoSystem` and setting dirty to false that will modify the debug component based on `sceneLODInfo.CurrentLOD`

File: Explorer/Assets/DCL/LOD/Systems/ResolveSceneLODInfo.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using ECS.StreamableLoading.AssetBundles;
+using ECS.StreamableLoading.Common.Components;
+using UnityEngine;
+using Utility;
+
+namespace DCL.LOD.Systems
+{
+    [UpdateInGroup(typeof(RealmGroup))]
+    [UpdateAfter(typeof(UpdateVisualSceneStateSystem))]
+    public partial class ResolveSceneLODInfo : BaseUnityLoopSystem
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/LOD/Systems/ResolveSceneLODInfo.cs:21]: Delete?

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/ISceneFacadeExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace SceneRunner.Scene
+{
+    public static class ISceneFacadeExtensions
+    {
+        public static void DisposeSceneFacadeAndRemoveFromCache(this ISceneFacade sceneFacade, IScenesCache scenesCache,
+            IReadOnlyList<Vector2Int> parcels)
+        {
+            //TODO: Misha, DisposeAsync is callen in many places. Should all of them add the parcels back to the cache?
+            sceneFacade.DisposeAsync().Forget();
```
Comment (mikhail-dcl) [inline Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/ISceneFacadeExtensions.cs:14]: The only other place is `RealmController`, right? We don't need to individually clean the cache in that case: it's handed by `scenesCache.Clear`:
```
private void FindLoadedScenes(World world)
        {
            allScenes.Clear();
            allScenes.AddRange(scenesCache.Scenes);

            // Dispose all scenes
            scenesCache.Clear();

            // find all loaded scenes
            world.Query(in SCENES, (ref ISceneFacade scene) => allScenes.Add(scene));
        }
```

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]
Everything loosk great: the only concern I have is regarding debugging flow of LODs

**Review by anicalbano — APPROVED** [review]
✔️ Reviewed and approved by QA

https://github.com/decentraland/unity-explorer/assets/139157886/b99a0810-60a7-474c-b147-3c6a7f1a85ac

File: Explorer/Assets/DCL/LOD/LODAsset.cs (hunk 1 above)
Comment (dalkia) [reply Explorer/Assets/DCL/LOD/LODAsset.cs]: I followed your inspiration, but did it a little bit different.

I did a system where every debug settings is encapsulated. Its a little bit expensive to turn on and to keep it running when its on due to some checks I got to do, but when tis off it has 0 effect. 

WDYT?

image
image
image

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/ISceneFacadeExtensions.cs (hunk 3 above)
Comment (mikhail-dcl) [reply Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/ISceneFacadeExtensions.cs:13]: Yes, this flow is for testing only so we don't need any kind of cache there

File: Explorer/Assets/DCL/LOD/LODAsset.cs (hunk 1 above)
Comment (dalkia) [reply Explorer/Assets/DCL/LOD/LODAsset.cs]: Removed allocations alltogether

**Review by mikhail-dcl — APPROVED** [review]

**Pull request merged**